src/realizations/catchment/Bmi_Multi_Formulation.cpp

```diff
@@ -306,23 +306,15 @@ string Bmi_Multi_Formulation::get_output_line_for_timestep(int timestep, std::st
         // This almost certainly should never happen, but just to be safe ...
         if (output_var_names.empty()) { return ""; }
 
-        try {
-            // Do the first separately, without the leading comma
-            auto output_data_provider_iter = availableData.find(output_var_names[0]);
-            *output_text_stream << get_var_value_as_double(output_var_names[0]);
-
-            // Do the rest with a leading comma
-            for (int i = 1; i < output_var_names.size(); ++i) {
-                *output_text_stream << "," << get_var_value_as_double(output_var_names[i]);
-            }
-            return output_text_stream->str();
-        }
-        catch (const std::exception &e) {
-            std::cerr << "WARN: " << e.what()
-                      << "; reverting to default behavior for multi-BMI formulation type (using last module)";
-            output_text_stream->str(std::string()); // ... clear any output contents being staged ...
-            is_out_vars_from_last_mod = true;          // ... revert to default behavior (just use last nested module)
+        // Do the first separately, without the leading comma
+        auto output_data_provider_iter = availableData.find(output_var_names[0]);
```

Comment:
This exists in the original code, but `output_data_provider_iter` is not actually tested or used through the rest of this function. That makes me concerned that the overall logic is (still) flawed.

Reply:
The ```output_data_provider_iter``` was actually used in ```check_output_var_names()``` function which was originally called from within this function. After unit test, the call to ```check_output_var_names()``` was moved to line 142 which is probably a more appropriate place.

Reply:
The `.find()` may have been left there to trigger the catch block that was removed... but in any case it's leftover code from 1a29230 that did the digging for each value before `get_var_value_as_double` existed, it appears. I'll drop this real quick since it's dead code and also not cost-free.